lib/yabeda/activejob.rb

```diff
@@ -97,7 +97,7 @@ def self.job_latency(event)
       return nil unless enqueue_time.present?
 
       enqueue_time = Time.parse(enqueue_time).utc unless enqueue_time.is_a?(Time)
-      perform_at_time = Time.parse(event.end.to_s).utc
+      perform_at_time = Time.at(event.end).utc
       (perform_at_time - enqueue_time)
```

Comment:
i have checked `ActiveSupport::Notifications::Event` and I can see that it exposes [duration method](https://api.rubyonrails.org/classes/ActiveSupport/Notifications/Event.html#method-i-duration)

I wonder if it would make sense to use that rather than having to compute the difference between start and end in here.

Reply:
The duration method returns end-start of the event itself I believe, where the existing calculation is end - enqueue_time (from the job embedded in the payload of the event), so I do not think using duration will return the same result.

If the event sat on the queue for a period of time before being performed then the start time of the event is likely to be quite a bit after the enqueued_at time held by the payload.